### ISSUE TITLE
Reorder properties

### ISSUE THREAD

**Issue by pri-kise** [anzwdev/al-code-outline#93]
I love the new function to add the application area property, but since the application area gets inserted at the end of my properties I would love some kind of reordering the properties to a specified order.

In the old world, they had always the same order, since it was a table and all values were always present.

I don't how if this can be solved, but if you could provide an option to oder the properties by alphabet and later providing some way of an individual order that would really help to find the specified porperties afterwards.

Eg. 
```
Caption = 'Documents';
Enabled = false;
Visible = false;
Image = Archive;
ApplicationArea = All;
```
is getting rearranged to :

```
ApplicationArea = All;
Caption = 'Documents';
Enabled = false;
Image = Archive;
Visible = false;
```

**Comment by anzwdev** [comment]
I've released a new version of the extension, it adds new "Sort properties" code action available if cursor is on the first line of object definition. When invoked, it will sort properties of every element defined inside the object.

**Comment by duichwer** [comment]
Works like a charm. I like it to have the ability to reorder manually, but maybe for future development some sort of auto-reorder trigger OnSave would be perfect. Is it possible to trigger the ordering after the formatting?

**Comment by rvanbekkum** [comment]
@duichwer @anzwdev Yes, that might be useful. Maybe you can add a setting that allows you to configure which code-actions you'd like to be triggered 'on-save'?
This setting could take a list or dictionary of code-action IDs, e.g., something like:
```json
                "alOutline.codeActionsOnSave": {
                    "type": "array",
                    "items": {
                        "type": "string",
                        "enum": [
                            "AA0008",
                            "AA0137",
                            "AA0139",
                            "SortDataItemColumns",
                            "SortProcedures",
                            "SortProperties"
                        ],
                        "enumDescriptions": [
                            "<description for AA0008>",
                            "<description for AA0137>",
                            "<description for AA0139>",
                            "<description for SortDataItemColumns>",
                            "<description for SortProcedures>",
                            "<description for SortProperties>"
                        ]
                    },
                    "default": [
                    ],
                    "uniqueItems": true,
                    "description": "Specifies which code-actions should be applied when a file is saved.",
                    "scope": "resource"
                },
```

**Comment by anzwdev** [comment]
That's a good idea, I'll add it

**Comment by anzwdev** [comment]
I had to redesign code actions functionality to be able to run sort procedures, properties and report colums actions on document save. It works now and is available in the latest extension update in Visual Studio Marketplace. I've added a new "alOutline.codeActionsOnSave" string array setting, it only supports 3 values now: "SortProperties", "SortProcedures", "SortReportColumns". To make it work, you also have to edit standard "editor.codeActionsOnSave" vs code setting. To run all 3 sorting code actions on document save, vs code settings should look like here:
```javascript
"editor.codeActionsOnSave": {
  "source.fixAll": true
},
"alOutline.codeActionsOnSave": [
  "SortProcedures", "SortProperties", "SortReportColumns"
]
```

Code actions that run on document save are no longer available in the quick fix menu.
